src/pages/build-apps/guides/transaction-signing.md

```diff
@@ -0,0 +1,260 @@
+  },
+  finished: data => {
+    console.log('Transaction ID:', data.txId);
+    console.log('Raw transaction:', data.txRaw);
+  },
+};
+
+await openContractCall(options);
+```
+
+Several parameters are available for calling `openContractCall`. Here's the exact interface for them:
+
+```tsx
+interface ContractCallOptions {
+  contractAddress: string;
+  functionName: string;
+  contractName: string;
+  functionArgs?: any[];
+  authOrigin?: string;
+  appDetails: {
+    name: string;
+    icon: string;
+  };
+  finished: (data: FinishedTxData) => void;
+}
+```
+
+| parameter       | type     | required | description                                                                                                                                                                                          |
+| --------------- | -------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| contractAddress | string   | true     | Stacks address to which contract is deployed                                                                                                                                                         |
+| contractName    | string   | true     | Name of contract to sign                                                                                                                                                                             |
+| functionName    | string   | true     | Name of function for signing / execution, which needs to be a [public function](/references/language-functions#define-public).                                                                       |
+| functionArgs    | array    | true     | Arguments for calling the function. [Provide the Clarity type with each argument](#passing-clarity-types-with-function-arguments). Defaults to `[]`.                                                 |
+| appDetails      | object   | true     | Dictionary that requires `name` and `icon` for app                                                                                                                                                   |
+| finished        | function | true     | Callback executed by app when transaction has been signed and broadcasted. It received an object back with `txId` and `txRaw` properties, both of which are strings.                                 |
+| authOrigin      | string   | false    | URL of authenticator to use for prompting signature and broadcast. Defaults to `https://wallet.hiro.so` for the Stacks Wallet, which is handled by the Stacks Wallet browser extension if installed. |
+
+### Passing Clarity types with function arguments
+
+To serialize your transaction properly, you'll need to provide an appropriate Clarity type for each function argument.
+
+These types are named the same as in Clarity and must be passed as strings:
```

Comment:
probably best to link to: `/references/language-types`

maybe also explain how this is different to samples in: `/write-smart-contracts/values`. Readers might get confused